src/test/protocol/PublicKey_test.cpp

```diff
@@ -433,32 +433,11 @@ class PublicKey_test : public beast::unit_test::suite
         testBase58(KeyType::ed25519);
     }
 
-    void
-    testMiscOperations()
-    {
-        testcase("Miscellaneous operations");
-
-        auto const pk1 = derivePublicKey(
-            KeyType::secp256k1,
-            generateSecretKey(
-                KeyType::secp256k1, generateSeed("masterpassphrase")));
-
-        PublicKey pk2(pk1);
-        BEAST_EXPECT(pk1 == pk2);
-        BEAST_EXPECT(pk2 == pk1);
-
-        PublicKey pk3;
-        pk3 = pk2;
-        BEAST_EXPECT(pk3 == pk2);
-        BEAST_EXPECT(pk1 == pk3);
```

Comment:
I like these tests and I don't think we should remove them. They verify several things:
- the copy constructor and assignment operators; and
- the equality operator.

If anything, I think they ought to be expanded to cover:
- things like self-assignment and self-move-assignment; and
- other comparison operators.

Reply:
I agree with your third and fourth bullet point about self assignment.
But do we need these tests even after having static_asserts for checking copy-construction, copy-assignment etc?

Reply:
> I agree with your third and fourth bullet point about self assignment. But do we need these tests even after having static_asserts for checking copy-construction, copy-assignment etc?

I was thinking about this same issue. The static asserts, AFAIK, only check that the functions _exist_. Actually doing the copy, assignment, etc. allows us to verify that the functions _work correctly_, which is way more important.

Reply:
oh interesting. I was under the assumption that static_asserts also check for the correctness of the function. 

In that case, we will need these unit tests even after having static_asserts.

I'll make the necessary changes, thanks.